app/controllers/assessments_controller.rb

```diff
@@ -193,6 +190,11 @@ def importAsmtFromTar
           FileUtils.chmod entry.header.mode, entry_file,
                           verbose: false
         elsif entry.file?
+          # Skip config files
+          next if existing_asmt && %W[#{asmt_name}.yml
```

Comment:
Nit: this seems a bit clunky since we won't upload files that have a matching basename but at a different path. e.g. `newdir/{asmt_name}.rb`

suggest comparing against `existing_asmt.path { ... filename ... }`, e.g. `existing_asmt.path "#{asmt_name}.yml"`. Even better to define helper methods in `assessment.rb` to return these paths (just like `handout_path`, `handin_directory_path`, `writeup_path`).

Note:
- yml: `dump_yaml` and `load_yaml` both use `path("#{name}.yml")` directly. Consider extracting into a helper method with definition `path "#{name}.yml"`
- rb: there exists `existing_asmt.unique_source_config_file_path`, but its definition could be refactored to `path "#{name}.rb"`
- log: currently `course_logger.rb` uses `Rails.root.join("courses", @course.name, @assessment.name, "log.txt")`. Consider extracting into a helper method with definition `path "log.txt"`